src/app_service/service/contacts/service.nim

```diff
@@ -601,6 +601,19 @@ QtObject:
     let data = ContactsStatusUpdatedArgs(statusUpdates: @[currentUserStatusUpdate])
     self.events.emit(SIGNAL_CONTACTS_STATUS_UPDATED, data)
 
+
+  proc markAsTrusted*(self: Service, publicKey: string) =
+    let response = status_contacts.markAsTrusted(publicKey)
+    if not response.error.isNil:
+      error "error marking as trusted ", msg = response.error.message
+      return
+
+    if self.contacts.hasKey(publicKey):
```

Comment:
I feel like this should error out if the `publicKey` is not an actual contact and not emit the signal below

Reply:
if there is no contact with provided `publicKey`, `status_contacts.markAsTrusted` will raise an error. This check is needed just in case there is no cached contact in nim layer